stix_shifter_modules/vision_one/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,301 @@
+import datetime
+import json
+import logging
+import re
+import time
+
+from stix_shifter_utils.stix_translation.src.json_to_stix import observable
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+from stix_shifter_utils.stix_translation.src.utils.transformers import TimestampToMilliseconds
+
+from .visionone_util import remove_enclosed_quote
+
+REFERENCE_DATA_TYPES = {
+    "dst": ["ipv4", "ipv6", "ipv4_cidr", "ipv6_cidr"],
+    "src": ["ipv4", "ipv6", "ipv4_cidr", "ipv6_cidr"],
+    "objectIp": ["ipv4", "ipv6", "ipv4_cidr", "ipv6_cidr"],
+    "sender": ["email-address"],
+    "source_ip": ["ipv4", "ipv6", "ipv4_cidr", "ipv6_cidr"],
+}
+START_STOP_STIX_QUALIFIER = r"START((t'\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z')|(\s\d{13}\s))STOP"
+TIMESTAMP = r"^'\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z'$"
+TIMESTAMP_MILLISECONDS = r"\.\d+Z$"
+
+logger = logging.getLogger(__name__)
+
+
+class QueryStringPatternTranslator:
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "AND",
+        ComparisonExpressionOperators.Or: "OR",
+        ComparisonComparators.Equal: ":",
+        ComparisonComparators.NotEqual: ":",
+        ComparisonComparators.Like: ":",
+        ObservationOperators.Or: 'OR',
+        ObservationOperators.And: 'AND'
+    }
+
+    def __init__(self, pattern: Pattern, data_model_mapper):
+        self.dmm = data_model_mapper
+        self.pattern = pattern
+        self.qualified_queries = []
+        self.translated = self.parse_expression(pattern, data_model_mapper.dialect)
+        self.qualified_queries.append(self.translated)
+        self.qualified_queries = _format_translated_queries(self.qualified_queries)
+
+    @staticmethod
+    def _format_equality(value) -> str:
+        # return '\'{}\''.format(value)
+        return '\"{}\"'.format(value)
+
+    @staticmethod
+    def _format_like(value) -> str:
+        # value = "'%{value}%'".format(value=value)
+        value = "{value}".format(value=value)
+        return QueryStringPatternTranslator._escape_value(value)
+
+    # pylint: disable=unused-argument
+    @staticmethod
+    def _escape_value(value, comparator=None) -> str:
+        if isinstance(value, str):
+            return '{}'.format(value.replace('\\', '\\\\').replace('\"', '\\"').replace('(', '\\(').replace(')', '\\)'))
```

Comment:
there are lots of unnecessary `\` in the translated query. you may want to update the method with something like this depending on your native query:
`return '{}'.format(value.replace('\"', '\\"').replace('(', '\\(').replace(')', '\\)'))
`

Reply:
Escaping `\` by `\\` is necessary for native query string of Trend Micro Vision One API.